### PR TITLE
kernel: EffectTS-inspired 'pipe' methods

### PR DESCRIPTION
**Pull request by fwbrasil**
This PR introduces overloaded `pipe` methods for up to six functions similarly to EffectTS but it's a method on the pending type, not in a companion object https://effect.website/docs/guides/essentials/pipeline

### CODE REVIEW

File: kyo-core/shared/src/main/scala/kyo/Clock.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ object Clock:
 
     private val local = Local.init(live)
 
-    def let[A, S](c: Clock)(f: => A < (IO & S))(using Frame): A < (IO & S) =
```
Comment (fwbrasil) [inline kyo-core/shared/src/main/scala/kyo/Clock.scala:16]: Minor unrelated cleanup. This used to require `IO` because `Local` was implemented based on it but it's not the case anymore.

**Review by hearnadam — APPROVED** [review]
LGTM, though this is not my favorite style 🙂 

It may be worth adding a combinator alias `|>`
